signup/tool/src/webapp/signup/organizer/userDefTsBlocks.jsp

```diff
@@ -78,23 +78,30 @@
 			function delayedRecalcDate(){
 				if (!wait){
 						wait = true;
-						for(i=0; i<30; i++){//control 30 ts
-						 setCustomEndtimeMonthDateYear(i);
-						}
-					  	setTimeout("wait=false;", 1500);//1.5 sec
-					}			
+						jQuery('input').each( function(index, data) {
+							var inputId = this.id;
+							if (inputId !== '' && inputId.endsWith('startTime')) {
+								// Get the input position
+								var inputPosition = inputId.replace('meeting:userDefinedTS:', '').replace(':startTime', '');
+								setCustomEndtimeMonthDateYear(inputPosition);
+							}
+							setTimeout("wait=false;", 1500);//1.5 sec
+						});
+					}
 			}
-				
-		</script>
-		<script type="text/javascript">			
+
 			jQuery(document).ready(function(){
+				jQuery('input').each( function(index, data) {

```

Comment:
I'd add a class in the jsf code and sleect on that, rather than just input. Makes it more robust.

Reply:
done